zebra-chain/src/transaction.rs

```diff
@@ -112,6 +112,10 @@ pub enum Transaction {
         inputs: Vec<transparent::Input>,
         /// The transparent outputs from the transaction.
         outputs: Vec<transparent::Output>,
+        /// The shielded data for this transaction, if any.
+        shielded_data: Option<ShieldedData>,
+        /// The net value of Sapling spend transfers minus output transfers.
+        value_balance: Amount,
```

Comment:
v5 transactions also include the Orchard value balance.

So we'll need to rename `shielded_data`, `ShieldedData`, and `value_balance` to include `sapling`. We should make this change across `V4` and `V5` transactions, for consistency.

Let's make this change in a follow-up PR, can you please open a ticket?

Reply:
Actually, we'll need to make this change in this PR, because the v5 transaction format changes the fields inside `ShieldedData`. So we'll need to:
* rename `ShieldedData` to `SaplingV4ShieldedData`
* add `SaplingV5ShieldedData`
* in a later PR, add `OrchardActionData`

@dconnolly what's a good name for the new `SaplingV5ShieldedData` type?
It doesn't have any proofs or signatures, just value commitments, nullifiers, keys, and ciphertext.

<img width="851" alt="Screen Shot 2021-03-08 at 09 02 55" src="https://user-images.githubusercontent.com/8951843/110258409-34d12d80-7fee-11eb-813f-4fa9387fb6e7.png">
<img width="851" alt="Screen Shot 2021-03-08 at 09 02 39" src="https://user-images.githubusercontent.com/8951843/110258411-3864b480-7fee-11eb-9f52-d0f89b4f3d93.png">

Reply:
What about having `ShieldedData::V4` and `ShieldedData::V5` instead ?

Reply:
> What about having `ShieldedData::V4` and `ShieldedData::V5` instead ?

That's still easily confused with Orchard shielded data. (And a bit confusing with Sprout shielded data as well.)